### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Opposites

### PR DESCRIPTION
**Pull request by mo271**
- [x] depends on: #1287
~~- [x] depends on tactic tidy?~~
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#1287
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by mattrobball** [comment]
`tidy` is being replaced by `aesop_cat`. I tested this on your branch and found it to work for all but a few times.

**Comment by mo271** [comment]
> `tidy` is being replaced by `aesop_cat`. I tested this on your branch and found it to work for all but a few times.

Great, I will use that later (in case you don't want to add a commit with those first...)

**Comment by mo271** [comment]
TODO:
- [ ] two failed applications of `ext`, perhaps we need specific ext lemmas here?
- [ ] last def `leftOpRightOpEquiv` 
- [ ] Naming

**Comment by mattrobball** [comment]
Also forgot: `aesop_cat` in place of `ext` too. Lean was very, very fussy with `leftOpRIghtOpEquiv`.

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1287~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by mo271** [comment]
merged with current HEAD

File: Mathlib/CategoryTheory/Opposites.lean [hunk 1]
```diff
@@ -0,0 +1,669 @@
...
+def leftOpRightOpEquiv : (Cᵒᵖ ⥤ D)ᵒᵖ ≌ C ⥤ Dᵒᵖ
+    where
+  functor :=
+    { obj := fun F => F.unop.rightOp
+      map := fun η => NatTrans.rightOp η.unop 
+    }
+  inverse :=
+    { obj := fun F => op F.leftOp
+      map := fun η => η.leftOp.op 
+    }
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Opposites.lean:655]:
```suggestion
  functor :=
    { obj := fun F => F.unop.rightOp
      map := fun η => NatTrans.rightOp η.unop  }
  inverse :=
    { obj := fun F => op F.leftOp
      map := fun η => η.leftOp.op }
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Comment by jcommelin** [comment]
bors f+

**Pull request closed without merging**